lib/web_ui/lib/src/engine/html/path/path_ref.dart

```diff
@@ -173,9 +182,47 @@ class PathRef {
   }
 
   /// Reconstructs Rect from path commands.
-  ui.Rect _getRect() {
-    return ui.Rect.fromLTRB(
-        atPoint(0).dx, atPoint(0).dy, atPoint(1).dx, atPoint(2).dy);
+  ui.Rect? _detectRect() {
+    assert(_fVerbs[0] == SPath.kMoveVerb);
+    final double x0 = atPoint(0).dx;
+    final double y0 = atPoint(0).dy;
+    final double x1 = atPoint(1).dx;
+    final double y1 = atPoint(1).dy;
+    if (_fVerbs[1] != SPath.kLineVerb || y1 != y0) {
+      return null;
+    }
+    final double width = x1 - x0;
+    final double x2 = atPoint(2).dx;
+    final double y2 = atPoint(2).dy;
+    if (_fVerbs[2] != SPath.kLineVerb || x2 != x1) {
+      return null;
+    }
+    final double height = y2 - y1;
+    final double x3 = atPoint(3).dx;
+    final double y3 = atPoint(3).dy;
+    if (_fVerbs[3] != SPath.kLineVerb || y3 != y2) {
+      return null;
+    }
+    if ((x2 - x3) != width || (y3 - y0) != height) {
+      return null;
+    }
+    return ui.Rect.fromLTWH(x0, y0, width, height);
+  }
+
+  /// Returns horizontal/vertical line bounds or null if not a line.
+  ui.Rect? getLine() {
```

Comment:
I'd give it a more specific name. When I read the call site I was immediately confused "What if the line is neither horizontal nor vertical?"

Reply:
Done.